src/qibolab/instruments/zhinst.py

```diff
@@ -0,0 +1,1050 @@
+                    )
+                )
+        if sweeper.parameter is Parameter.amplitude:
+            for pulse in sweeper.pulses:
+                pulse.amplitude = pulse.amplitude * max(sweeper.values)
+                sweeper.values = sweeper.values / max(sweeper.values)
+                parameter = ZhSweeper(pulse, sweeper, qubits[sweeper.pulses[0].qubit]).zhsweeper
+
+        if sweeper.parameter is Parameter.bias:
+            for qubit in sweeper.qubits:
+                parameter = ZhSweeperLine(sweeper, qubit, self.sequence_qibo).zhsweeper
+
+        elif sweeper.parameter is Parameter.delay:
+            parameter = ZhSweeperLine(sweeper).zhsweeper
+
+        elif parameter is None:
+            parameter = ZhSweeper(sweeper.pulses[0], sweeper, qubits[sweeper.pulses[0].qubit]).zhsweeper
+
+        with exp.sweep(
+            uid=f"sweep_{sweeper.parameter.name.lower()}_{i}",  # FIXME: This uid trouble double freq ???
+            parameter=parameter,  # FIXME: This uid for double freq ???
+            reset_oscillator_phase=True,  # FIXME: Should we reset this phase ???
+        ):
+            if len(self.sweepers) > 0:
+                self.sweep_recursion(qubits, exp, exp_calib, exp_options)
+            else:
+                self.select_exp(exp, qubits, exp_options)
+
+    def sweep_recursion_NT(self, qubits, options, exp, exp_calib):
+        """
+        Sweepers recursion for Near Time sweepers. Faster than regular software sweepers as
+        they are executed on the actual device by (software ? or slower hardware ones)
+
+        You want to avoid them so for now they are implement for a specific sweep.
+        """
+
+        print("NT Loop")
+
+        sweeper = self.NT_sweeps[0]
+
+        i = len(self.NT_sweeps) - 1
+        self.NT_sweeps.remove(sweeper)
+
+        parameter = None
+
+        if sweeper.parameter is Parameter.amplitude:
+            for pulse in sweeper.pulses:
+                line = "drive" if pulse.type is PulseType.DRIVE else "measure"
+                zhsweeper = ZhSweeper(pulse, sweeper, qubits[sweeper.pulses[0].qubit]).zhsweeper
+                zhsweeper.uid = f"amplitude"
+                path = "DEV12146"  # Hardcoded for SHFQC(SHFQA)
+
+        # Leave it for dual freq if they dont work in RT
+        if sweeper.parameter is Parameter.frequency:
+            for pulse in sweeper.pulses:
+                line = "drive" if pulse.type is PulseType.DRIVE else "measure"
+                zhsweeper = ZhSweeper(pulse, sweeper, qubits[sweeper.pulses[0].qubit]).zhsweeper
+                zhsweeper.uid = f"frequency"  # TODO: Changing the name from "frequency" breaks it
+                exp_calib[f"{line}{pulse.qubit}"] = lo.SignalCalibration(
+                    oscillator=lo.Oscillator(
+                        frequency=zhsweeper,
+                        modulation_type=lo.ModulationType.HARDWARE,
+                    )
+                )
+
+        elif parameter is None:
```

Comment:
Here the `parameter` is used, but has not been modified before, so the condition should be always `True`

```suggestion
        else:
```

Reply:
If this is the case, move the `parameter = None` in the `if` branch, just to the line above this one